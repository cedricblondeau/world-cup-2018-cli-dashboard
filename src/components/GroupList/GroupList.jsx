import React from 'react';
import PropTypes from 'prop-types';

import GroupDetails from './GroupDetails/GroupDetails';

const GroupList = props => {
  const groupWidth = 22;
  const groupCount = 8;

  const groups = props.groups.map((group, i) => (
    <GroupDetails
      group={group.group}
      key={`group-details-${group.group.id}`}
      left={i * groupWidth}
      width={groupWidth}
    />
  ));

  return (
    <element height={6} width={groupWidth * groupCount} left="center">
      {groups}
    </element>
  );
};

GroupList.propTypes = {
  groups: PropTypes.array.isRequired,
};

export default GroupList;
