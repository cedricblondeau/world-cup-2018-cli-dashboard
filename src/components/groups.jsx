import axios from 'axios';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Group from './group';

class Groups extends Component {
  constructor(props) {
    super(props);

    this.state = { groups: null };
  }

  componentDidMount() {
    this.updateGroups();
  }

  get groups() {
    if (!this.state.groups) {
      return 'Fetching...';
    }

    return this.state.groups.map((group, i) => {
      const x = i % 2;
      const y = Math.floor(i / 2);

      return (
        <Group
          key={`group-${group.group.id}`}
          group={group.group}
          left={`${x * 50}%`}
          top={`${y * 25}%`}
        />
      );
    });
  }

  async updateGroups() {
    try {
      const response = await axios.get(
        'http://worldcup.sfg.io/teams/group_results',
      );
      if (!Array.isArray(response.data)) {
        this.props.debug(`Groups - Received unexpected data: ${response.data}`);
        return;
      }

      this.setState({ groups: response.data });
    } catch (e) {
      this.props.debug(`Groups - ${e.message}`);
    }
  }

  render() {
    return (
      <element width="45%" height="100%" top="0" left="55%" border="line">
        <element left={4} top={1}>
          {this.groups}
        </element>
      </element>
    );
  }
}

Groups.propTypes = {
  debug: PropTypes.func.isRequired,
};

export default Groups;
