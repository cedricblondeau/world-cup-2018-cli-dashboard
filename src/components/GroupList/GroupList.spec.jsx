import React from 'react';
import TestRenderer from 'react-test-renderer';

import groups from '../../../fixtures/group_results.json';
import GroupList from './GroupList';

describe('<GroupList>', () => {
  it('Renders properly', () => {
    const groupListRendered = TestRenderer.create(
      <GroupList groups={groups} />,
    );
    expect(groupListRendered.toJSON()).toMatchSnapshot();
  });
});
