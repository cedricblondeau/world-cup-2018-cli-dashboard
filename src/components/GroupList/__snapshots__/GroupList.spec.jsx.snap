// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupList> Renders properly 1`] = `
<element
  height={6}
  left="center"
  width={144}
>
  <element
    height={6}
    left={0}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group A{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[34mâ– [39m URU 9
[37mâ– [39m RUS 6
[32mâ– [39m KSA 3
[30mâ– [39m EGY 0
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={18}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group B{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[31mâ– [39m ESP 5
[32mâ– [39m POR 5
[32mâ– [39m IRN 4
[31mâ– [39m MAR 1
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={36}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group C{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[34mâ– [39m FRA 7
[31mâ– [39m DEN 5
[31mâ– [39m PER 3
[33mâ– [39m AUS 1
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={54}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group D{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[31mâ– [39m CRO 9
[34mâ– [39m ARG 4
[32mâ– [39m NGA 3
[34mâ– [39m ISL 1
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={72}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group E{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[33mâ– [39m BRA 7
[31mâ– [39m SUI 5
[31mâ– [39m SRB 3
[31mâ– [39m CRC 1
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={90}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group F{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[33mâ– [39m SWE 6
[32mâ– [39m MEX 6
[31mâ– [39m KOR 3
[32mâ– [39m GER 3
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={108}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group G{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[31mâ– [39m BEL 9
[31mâ– [39m ENG 6
[31mâ– [39m TUN 3
[34mâ– [39m PAN 0
"
      height={4}
      left="center"
      top={2}
    />
  </element>
  <element
    height={6}
    left={126}
    top={0}
    width={18}
  >
    <box
      align="center"
      content="{bold}Group H{/bold}"
      height={1}
      tags={true}
    />
    <box
      content="[33mâ– [39m COL 6
[34mâ– [39m JPN 4
[32mâ– [39m SEN 4
[31mâ– [39m POL 3
"
      height={4}
      left="center"
      top={2}
    />
  </element>
</element>
`;
